xDrip/Components/Charts/SupportingLogic/GlucoseChartProvider.swift

```diff
@@ -46,5 +71,175 @@ extension GlucoseChartProvider where Self: UIView {
             context.setFillColor(color.cgColor)
             context.fillEllipse(in: circleRect)
         }
+        
+        drawBasalStroke()
+    }
+    
+    func drawBasalStroke() {
+        guard let context = UIGraphicsGetCurrentContext() else { return }
+        guard let displayMode = User.current.settings.chart?.basalDisplayMode, displayMode != .notShown else { return }
+        let yIntervalBasal = yInterval / 4.0
+        let yMax = displayMode == .onBottom ? yInterval - yIntervalBasal : 0
+        let yMin = displayMode == .onBottom ? yInterval : yIntervalBasal
+        
+        guard var max = basalEntries.map({ $0.value }).max() else { return }
```

Comment:
This is not an effective way to find max value. Use `basalEntries.max(by: )` function.